DuckDuckGo/Autoconsent/AutoconsentUserScript.swift

```diff
@@ -291,13 +290,44 @@ extension AutoconsentUserScript {
         }
     }
 
+    @MainActor
+    func handlePopupFound(message: WKScriptMessage, replyHandler: @escaping (Any?, String?) -> Void) {
+        guard let messageData: PopupFoundMessage = decodeMessageBody(from: message.body) else {
+            replyHandler(nil, "cannot decode message")
+            return
+        }
+        Logger.autoconsent.info("Cookie popup found: \(String(describing: messageData))")
+        guard let url = URL(string: messageData.url),
+              let host = url.host else {
+            replyHandler(nil, "cannot decode message")
+            return
+        }
+
+        if messageData.cmp == "filterList" {
+            refreshDashboardState(consentManaged: true, cosmetic: true, optoutFailed: false, selftestFailed: nil)
+            // trigger animation, but do not cache it because it can still be overridden
+            if !management.sitesNotifiedCache.contains(host) {
+                Logger.autoconsent.info("Starting animation for cosmetic filters")
+                // post popover notification on main thread
+                DispatchQueue.main.async {
```

Comment:
the comment is misleading since the method is already performed on the `MainActor`; 
and the `newSitePopupHiddenNotification` notification handler is dispatched asynchronously on the main queue as well - so the `DispatchQueue.main.async {` can be removed here.

Reply:
You mean `handleAutoconsentDone`? Yes, I think so, you can see it‘s handled at `NavigationBarViewController.swift:537` – `showAutoconsentFeedback` that‘s using `DispatchQueue.main.async`, and since the sending function is `@MainActor` i.e. handling the message on the main thread actor this seems as an unneeded code duplication, not critical though.